libr/debug/p/debug_native.c

```diff
@@ -1524,8 +1524,14 @@ static int r_debug_native_drx (RDebug *dbg, int n, ut64 addr, int sz, int rwx, i
 	regs[7] = r_reg_getv (R, "dr7");
 
 	if (sz == 0) {
-		drx_list ((drxt*)&regs);
-		return false;
+		if (addr != 0) {
```

Comment:
Invalid addr should be ut64max

Reply:
This comment led me to refactor the drx API. I think it's more sensible to send the type of operation as a parameter instead of deriving it from the regular parameters. Perhaps I should have done that in a different pull request though.

This added a bit more code to the pull request, do you need me to split the pull request / add tests?